sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1295,34 +1295,47 @@ def test_saga_vs_liblinear():
 
 
 @pytest.mark.parametrize('multi_class', ['ovr', 'multinomial'])
-@pytest.mark.parametrize('solver', ['newton-cg', 'saga'])
-def test_dtype_match(solver, multi_class):
-    # Test that np.float32 input data is not cast to np.float64 when possible
+@pytest.mark.parametrize('solver', ['newton-cg', 'liblinear', 'saga'])
+@pytest.mark.parametrize('fit_intercept', [False, True])
+def test_dtype_match(solver, multi_class, fit_intercept):
+    # Test that 32-bit input produces output of at least 32-bit precision, that
+    # 64-bit input produces output of 64-bit precision, and that the output is
+    # approximately the same no matter the input.
+
+    if solver == 'liblinear' and multi_class == 'multinomial':
+        return
 
     X_32 = np.array(X).astype(np.float32)
     y_32 = np.array(Y1).astype(np.float32)
     X_64 = np.array(X).astype(np.float64)
     y_64 = np.array(Y1).astype(np.float64)
     X_sparse_32 = sp.csr_matrix(X, dtype=np.float32)
+    X_sparse_64 = sp.csr_matrix(X, dtype=np.float64)
     solver_tol = 5e-4
 
     lr_templ = LogisticRegression(
         solver=solver, multi_class=multi_class,
-        random_state=42, tol=solver_tol, fit_intercept=True)
-    # Check type consistency
+        random_state=42, tol=solver_tol, fit_intercept=fit_intercept)
+
+    # Check 32-bit type consistency
     lr_32 = clone(lr_templ)
     lr_32.fit(X_32, y_32)
-    assert lr_32.coef_.dtype == X_32.dtype
+    assert lr_32.coef_.dtype in [np.float32, np.float64]
```

Comment:
Why this change? The whole point of adding support of 32bit in other solvers was to ensure that `coef_` is of the same dtype as `X`. We can't remove that constraint for other solvers at least.

Reply:
As I explained at https://github.com/scikit-learn/scikit-learn/pull/14296#issuecomment-510271098, there is no reason to prevent any solver from producing 64-bit output from 32-bit input.